Header header
bool ir0
bool ir1
bool ir2
bool cliff0
bool cliff1
